### PR TITLE
Glossary.event handlers

### PR DESCRIPTION
**Pull request by wxtim**
Added a definition of "event handlers" to glossary. I'd have wanted it as a user.

Associated with https://github.com/cylc/cylc-flow/pull/4290

### CODE REVIEW

File: src/7-to-8/major-changes/platforms.rst [hunk 1]
```diff
@@ -0,0 +1,98 @@
+Platforms
+=========
+
+.. note::
+
+   The terms :term:`platform` and :term:`job platform` are equivelent.
```
Comment (kinow) [inline src/7-to-8/major-changes/platforms.rst:6]: s/equivelent/equivalent

**Review by kinow — COMMENTED** [review]
Just confirming @wxtim , did you intentionally add a commit from #255 in this PR too?

File: src/7-to-8/major-changes/platforms.rst (hunk 1 above)
Comment (kinow) [reply src/7-to-8/major-changes/platforms.rst:6]: Started reviewing when I noticed this file had been changed in #255, see comment above :point_up:

Comment (wxtim) [reply src/7-to-8/major-changes/platforms.rst:6]: probably not.
Oops.

**Comment by wxtim** [comment]
> Just confirming @wxtim , did you intentionally add a commit from #255 in this PR too?

No, and I think that it's gone now.

File: src/glossary.rst [hunk 2]
```diff
@@ -1165,3 +1165,13 @@ Glossary
 
          It is also possible to have multiple :term:`flows <flow>` running in the
          scheduler :term:`schduler` simultaneously.
+
+   event handlers
+   handlers
+      Carry out an action when the Cylc scheduler detects that an arbitrary
+      event has occurred. This allows Cylc to centralize automated handling of
+      critical events. Cylc can do may things when it detects an event.
```
Comment (kinow) [inline src/glossary.rst:1173]: s/may/many

**Review by kinow — APPROVED** [review]
One typo (I think), otherwise looks good. I don't use event handlers, so maybe the second reviewer can review the description. Thanks @wxtim !

**Comment by oliver-sanders** [comment]
Important information to convey:

* These are "hooks" you can configure to do what you want when the event occurs.
* There are [scheduler] events (e.g. startup, stall, shjutdown) and [runtime][task] events (e.g. succeeded, submit-failed, failed).
* Links to the relevant configurations.

File: src/glossary.rst [hunk 3]
```diff
@@ -1165,3 +1165,13 @@ Glossary
 
          It is also possible to have multiple :term:`flows <flow>` running in the
          scheduler :term:`schduler` simultaneously.
+
+   event handlers
+   handlers
+      Carry out an action when the Cylc scheduler detects that an arbitrary
```
Comment (hjoliver) [inline src/glossary.rst:1171]:
```suggestion
      A command or action to execute when the Cylc scheduler detects that a particular 
```

File: src/glossary.rst [hunk 4]
```diff
@@ -1165,3 +1165,13 @@ Glossary
...
+      event has occurred. This allows Cylc to centralize automated handling of
+      critical events. Cylc can do may things when it detects an event.
+      Use-cases include:
+
+      - Send an email message
+      - Run a Cylc command
```
Comment (hjoliver) [inline src/glossary.rst:1177]:
```suggestion
      - Send an email message
      - Run a Cylc command
      - Run any user-specified script or command
```

**Comment by wxtim** [comment]
>     * Links to the relevant configurations.

I linked to the full Event Handling docs which has these links. I'm happy to copy and paste them, but I think that it's better not to: It's more to break...

**Comment by oliver-sanders** [comment]
> I'm happy to copy and paste them, but I think that it's better not to: It's more to break...

We've got the global referencing syntax which is subject to automatic integrity checking:

```rst
:cylc:conf:`flow.cylc[scheduler][events]`
:cylc:conf:`flow.cylc[runtime][<task>][events]`
```

**Comment by wxtim** [comment]
I think that I have now responded to your comments @oliver-sanders . This only needs to be a summary, because there's a fuller explanation elsewhere.

File: src/user-guide/writing-workflows/runtime.rst [hunk 5]
```diff
@@ -23,11 +23,20 @@ or for triggering other tasks off all members at once -
...
 Namespace Names
 ---------------
 
 .. autoclass:: cylc.flow.unicode_rules.TaskNameValidator
 
+
+.. note::
+
+   - ``:`` would stop cylc adding paths using task name to ``$PATH``.
+   - ``.`` would be interpreted as a namespace delimeter.
```
Comment (oliver-sanders) [inline src/user-guide/writing-workflows/runtime.rst:37]: I'm not sure what this means.

File: src/user-guide/writing-workflows/runtime.rst [hunk 6]
```diff
@@ -23,11 +23,20 @@ or for triggering other tasks off all members at once -
...
+
+
```
Comment (wxtim) [inline src/user-guide/writing-workflows/runtime.rst:39]:
```suggestion
```

**Review by oliver-sanders — APPROVED** [review]

**Pull request merged**